util/git/creds.go

```diff
@@ -65,7 +79,7 @@ type GenericHTTPSCreds interface {
 	HasClientCert() bool
 	GetClientCertData() string
 	GetClientCertKey() string
-	Environ() (io.Closer, []string, error)
+	Environ(store CredsStore) (io.Closer, []string, error)
```

Comment:
Instead of passing `store CredsStore` all the way down the call chain to each plugin implementation, such that it eventually makes its way into the Envrion() call, could we more simply add `store CredsStore` as a first-class field to HTTPSCreds / GitHubAppCreds whose `Environ()` will know to Add & Remove() the NONCE as part of the returned closer?

Reply:
Tried to implement your suggestion and it is indeed much better. Pushed commit that implements suggested change: https://github.com/argoproj/argo-cd/pull/8516/commits/52926c8f1d0d09e35518fc80003dcf79cd294633